src/serializers/type_serializers/datetime_etc.rs

```diff
@@ -2,21 +2,35 @@ use std::borrow::Cow;
 
 use pyo3::prelude::*;
 use pyo3::types::{PyDate, PyDateTime, PyDict, PyTime};
-use pyo3::IntoPyObjectExt;
 
+use super::{
+    infer_json_key, infer_serialize, infer_to_python, BuildSerializer, CombinedSerializer, Extra, SerMode,
+    TypeSerializer,
+};
 use crate::definitions::DefinitionsBuilder;
 use crate::input::{pydate_as_date, pydatetime_as_datetime, pytime_as_time};
+use crate::serializers::config::{FromConfig, TemporalMode};
 use crate::PydanticSerializationUnexpectedValue;
 
-use super::{
-    infer_json_key, infer_serialize, infer_to_python, py_err_se_err, BuildSerializer, CombinedSerializer, Extra,
-    SerMode, TypeSerializer,
-};
-
 pub(crate) fn datetime_to_string(py_dt: &Bound<'_, PyDateTime>) -> PyResult<String> {
     pydatetime_as_datetime(py_dt).map(|dt| dt.to_string())
 }
 
+pub(crate) fn datetime_to_seconds(py_dt: &Bound<'_, PyDateTime>) -> PyResult<f64> {
+    pydatetime_as_datetime(py_dt).map(|dt| dt.timestamp() as f64)
+}
+
+pub(crate) fn datetime_to_milliseconds(py_dt: &Bound<'_, PyDateTime>) -> PyResult<f64> {
+    pydatetime_as_datetime(py_dt).map(|dt| dt.timestamp_ms() as f64)
+}
```

Comment:
These two seem wrong to me, probably need to go through the functions on `EitherTimedelta`?